src/ExchangeSharp/API/Exchanges/Coinbase/ExchangeCoinbaseAPI.cs

```diff
@@ -454,10 +454,22 @@ protected override async Task<ExchangeOrderResult> OnPlaceOrderAsync(ExchangeOrd
   			// The Post doesn't return with any status, just a new OrderId. To get the Order Details we have to reQuery.
   			return await OnGetOrderDetailsAsync(result[ORDERID].ToStringInvariant());
   		}
-  		catch (Exception ex) // All fails come back with an exception. 
+  		catch (Exception ex) // All fails come back with an exception.
   		{
+				Logger.Error(ex, "Failed to place coinbase error");
   			var token = JToken.Parse(ex.Message);
-  			return new ExchangeOrderResult(){ Result = ExchangeAPIOrderResult.Rejected, ClientOrderId = order.ClientOrderId, ResultCode = token["error_response"]["error"].ToStringInvariant() };
+  			return new ExchangeOrderResult(){
+					Result = ExchangeAPIOrderResult.Rejected,
+					IsBuy = payload["side"].ToStringInvariant().Equals(BUY),
+					MarketSymbol = payload["product_id"].ToStringInvariant(),
+					ClientOrderId = order.ClientOrderId,
+					ResultCode = $"{token["error_response"]["error"].ToStringInvariant()} - {token["error_response"]["preview_failure_reason"].ToStringInvariant()}",
+					AmountFilled = 0,
+					Amount = order.Amount,
+					AveragePrice = 0,
+					Fees = 0,
+					FeesCurrency = "USDT"
+				};
```

Comment:
include order details since the error handler uses `.ToString()` on the ExchangeOrderResults and without details it is useless. Also `token["error_response"]["error"]` seems insufficient in many cases so including `token["error_response"]["preview_failure_reason"]` as well